the-collab-lab/tcl-24-smart-shopping-list
# 1 cv kr connect firebase

## Description

This PR adds firebase react-firebase-hooks to the dependencies and Data, Form components as well as DataContext and custom hook named useFirebase.

- In Data.jsx we fetched the document information from the collection "things" and displayed on a list.
- In Form.jsx we made a form for add data to the collection
- In DataContext we created the new context and return the provider with the value returned from custom hook
- In useFirebase as a custom hook, we created functions getAll, create, update, remove and getById.
  
## Related Issue

Closes #2 

## Acceptance Criteria

* The following have been added as dependencies in the project: firebase and react-firebase-hooks
* Make changes in the Firestore database that are displayed in the app 
* Make changes in the app that are reflected on the database 

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
| ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

### After

![Captura de pantalla de 2021-04-07 23-26-04](https://user-images.githubusercontent.com/39011391/114031486-6cc7cc80-9849-11eb-8bee-29bf527403b4.png)


## Testing Steps / QA Criteria

- From your terminal, pull down the branch
- Run npm install to install the dependencies
- Run npm start to view the features in the browser
- Check the items on the collections
- Add an item to the collection
- Check firebase to see your changes
